back/node_watcher/src/nodeWatcher.ts

```diff
@@ -33,12 +33,21 @@ function waitFinalized(
 
 async function incrementor(
   api: ApiPromise,
+  provider: WsProvider,
   tasks: NomidotTask[]
 ): Promise<void> {
   let blockIndex = parseInt(process.env.START_FROM || '0');
   let currentSpecVersion = api.createType('u32', -1);
   let lastKnownBestFinalized = await waitFinalized(api, 0);
 
+  api.once('disconnected', async () => {
+    try {
+      api = await ApiPromise.create({ provider });
```

Comment:
ah, I don't like mutating function arguments. But i'm very curious if this works